src/interface.rs

```diff
@@ -44,6 +48,14 @@ fn enable_logs() {
     });
 }
 
+fn cleanup_mathml_with_rules<'a>(_rules: &SpeechRules, mathml: Element<'a>) -> Result<Element<'a>> {
+    // TODO: Canonicalization does not seem to actually use rules?
```

Comment:
I have a question about this: I haven't been able to find where canonicalization actually uses rules, but the comment here seems to indicate it does?

https://github.com/NSoiffer/MathCAT/blob/ca4b93399aa9b3a7cbe412cac1791d45bba553f4/src/interface.rs#L109-L111

Reply:
Canonicalization uses the preference files (e.g., for decimal and block separators to fix up numbers). It also reads the definition files so it knows what are function names (and other things I think) for restructuring the code.